bench/Chart.js3.html

```diff
@@ -0,0 +1,153 @@
+		<script src="https://www.chartjs.org/dist/master/Chart.min.js"></script>
+		<script src="https://cdn.jsdelivr.net/npm/chartjs-adapter-luxon@0.2.0"></script>
+		<script src="https://www.chartjs.org/samples/latest/utils.js"></script>
+
+		<div class="chart-container">
+			<canvas id="chart1"></canvas>
+		</div>
+
+		<script>
+			function round2(val) {
+				return Math.round(val * 100) / 100;
+			}
+
+			function round3(val) {
+				return Math.round(val * 1000) / 1000;
+			}
+
+			function prepData(packed) {
+				console.time('prep');
+
+				// epoch,idl,recv,send,read,writ,used,free
+
+				const numFields = packed[0];
+				packed = packed.slice(numFields + 1);
+
+				var cpu = Array(packed.length/numFields);
+				var ram = Array(packed.length/numFields);
+				var tcp = Array(packed.length/numFields);
+
+				for (let i = 0, j = 0; i < packed.length; i += numFields, j++) {
+					let date = packed[i] * 60 * 1000;
+					cpu[j] = {x: date, y: round3(100 - packed[i+1])};
+					ram[j] = {x: date, y: round2(100 * packed[i+5] / (packed[i+5] + packed[i+6]))};
+					tcp[j] = {x: date, y: packed[i+3]};
+				}
+
+				console.timeEnd('prep');
+
+				return [cpu, ram, tcp];
+			}
+
+			function makeChart(data) {
+				console.time('chart');
+
+				var ctx = document.getElementById('chart1').getContext('2d');
+				Chart.defaults.global.animation = false;
+				var color = Chart.helpers.color;
+				var cfg = {
+					data: {
+						datasets: [
+							{
+								label: 'CPU',
+								borderColor: window.chartColors.red,
+								data: data[0],
+								type: 'line',
+								pointRadius: 0,
+								fill: false,
+								lineTension: 0,
+								borderWidth: 1
+							},
+							{
+								label: 'RAM',
+								borderColor: window.chartColors.blue,
+								data: data[1],
+								type: 'line',
+								pointRadius: 0,
+								fill: false,
+								lineTension: 0,
+								borderWidth: 1
+							},
+							{
+								label: 'TCP Out',
+								borderColor: window.chartColors.green,
+								data: data[2],
+								type: 'line',
+								pointRadius: 0,
+								fill: false,
+								lineTension: 0,
+								borderWidth: 1
+							}
+						]
+					},
+					options: {
+						maintainAspectRatio: false,
+						parsing: false,
+						tooltips: {
+							mode: 'index',
+							intersect: false,
+						},
+						scales: {
+							x: {
+								type: 'time',
+								distribution: 'series',
+								ticks: {
+									source: 'auto',
+									maxRotation: 0,
+									autoSkip: true,
+									autoSkipPadding: 75,
+									sampleSize: 1
+								}
+							},
+							y: {
+								scaleLabel: {
+									display: true,
+									labelString: '%'
+								}
+							}
+						},
+					}
+				};
+
+				var chart = new Chart(ctx, cfg);
+
+				wait.textContent = "Done!";
+				console.timeEnd('chart');
+			}
+
+			let wait = document.getElementById("wait");
+			wait.textContent = "Fetching data.json (2.07MB)....";
+			fetch("data.json").then(r => r.json()).then(packed => {
```

Comment:
this line is a little annoying because it doesn't work on local. what do you think about changing all the samples to use `https://leeoniya.github.io/uPlot/bench/data.json` instead?

Reply:
> it doesn't work on local

i usually test all the demos and benches with `http-server`

> what do you think about changing all the samples to use https://leeoniya.github.io/uPlot/bench/data.json instead?

i'm frequently on a tethered connection and pumping 2mb of json with caching disabled in devtools would be a no-go for me (both slow and i'd burn through my cap 10x faster).